### PR TITLE
netgen-mesher: init at version 6.2.2004

### LINKED ISSUE #49532: Please add netgen package
Upstream site: https://ngsolve.org
https://sourceforge.net/projects/netgen-mesher/
Most other repos package it as 'netgen' see https://repology.org/metapackage/netgen/versions

### PR DESCRIPTION
**Pull request by Dema**
###### Motivation for this change

Wanted to try myself at creating nix packages. Closes #49532

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

`meta.maintainers` is NOT set.

### CODE REVIEW

File: pkgs/applications/science/geometry/netgen-mesher/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{stdenv, fetchgit, cmake, python3, opencascade, tcl, tk, libjpeg, zlib, mesa, libGLU, xorg, ffmpeg, pybind11 }: 
+
+stdenv.mkDerivation rec {
+  pname = "netgen-mesher";
+  
+  version="6.2.1909";
+
+  src = fetchgit {
```
Comment (risicle) [inline pkgs/applications/science/geometry/netgen-mesher/default.nix]: `fetchFromGitHub` is preferred for github repos. Many examples throughout nixpkgs to refer to.

File: pkgs/applications/science/geometry/netgen-mesher/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  nativeBuildInputs = [cmake];
+  
+  buildInputs= [python3 pybind11 tcl tk libjpeg ffmpeg zlib mesa libGLU	xorg.libXmu ];
+
+preConfigure = ''
+  cmakeFlagsArray=(
+  -DUSE_JPEG=ON
+  -DUSE_MPEG=ON
+  )
+'';
```
Comment (risicle) [inline pkgs/applications/science/geometry/netgen-mesher/default.nix]: You don't have to break out into bash for this - you should just

```
  cmakeFlags = [
    "-DUSE_JPEG=ON"
    "-DUSE_MPEG=ON"
  ];
```

File: pkgs/applications/science/geometry/netgen-mesher/default.nix (hunk 1 above)
Comment (Dema) [reply pkgs/applications/science/geometry/netgen-mesher/default.nix]: I tried, but turns out it uses git submodules, so fetchFromGitHub won't work here.

Comment (risicle) [reply pkgs/applications/science/geometry/netgen-mesher/default.nix]: Oh... I was under the impression `fetchgit` would only fetch submodules if `fetchSubmodules` was specifically set - but looks like it's default `true`.

Would be good to either explicitly set this to `true` or leave a comment with the reason for this.

Comment (Dema) [reply pkgs/applications/science/geometry/netgen-mesher/default.nix]: Ok! I'll add a comment. It has it's own version of pybind11 library in `external_dependencies/pybind11`. I tried to convince it to work with pybind11 from  nixpkgs, but failed. I couldn't force cmake to find it. Maybe I should try harder :smile:

File: pkgs/applications/science/geometry/netgen-mesher/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, opencascade, fetchFromGitHub, cmake, python3, tcl, tk, libjpeg, zlib, mesa, libGLU, xorg, ffmpeg }:
+
+
+stdenv.mkDerivation rec {
+  pname = "netgen-mesher";
+
+  version = "6.2.2004";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/geometry/netgen-mesher/default.nix:7]:
```suggestion

stdenv.mkDerivation rec {
  pname = "netgen-mesher";
  version = "6.2.2004";
```

File: pkgs/applications/science/geometry/netgen-mesher/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchFromGitHub {
+    owner = "NGSolve";
+    repo = "netgen";
+    rev = "v${version}";
+    # pybind11 is a submodule
+    fetchSubmodules = true;
+    sha256 = "05s982z1s3rny20cafwbrr7n1z0ql7czq97zs75hgqwrbldfij5y";
+  };
+
+  enableParallelBuilding = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/geometry/netgen-mesher/default.nix:18]:
```suggestion
```

cmake implies that.

File: pkgs/applications/science/geometry/netgen-mesher/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "-DUSE_OCC=ON"
+    "-DOCC_INCLUDE_DIR=${opencascade}/include/oce"
+  ];
+
+  meta = {
+    description = "An automatic 3d tetrahedral mesh generator";
+    homepage = "https://sourceforge.net/projects/netgen-mesher/";
+    license = stdenv.lib.licenses.lgpl21;
+    platforms = stdenv.lib.platforms.linux;
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/science/geometry/netgen-mesher/default.nix:36]:
```suggestion
  meta = with stdenv.lib; {
    description = "An automatic 3d tetrahedral mesh generator";
    homepage = "https://sourceforge.net/projects/netgen-mesher/";
    license = licenses.lgpl21;
    platforms = platforms.linux;
    Please add yourself as a maintainer.
  };
```

File: pkgs/applications/science/geometry/netgen-mesher/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    # pybind11 is a submodule
+    fetchSubmodules = true;
+    sha256 = "05s982z1s3rny20cafwbrr7n1z0ql7czq97zs75hgqwrbldfij5y";
+  };
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ python3 tcl tk libjpeg ffmpeg zlib mesa libGLU xorg.libXmu opencascade ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/geometry/netgen-mesher/default.nix:22]: Is python only a build time depedency?

File: pkgs/applications/science/geometry/netgen-mesher/default.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, opencascade, fetchFromGitHub, cmake, python3, tcl, tk, libjpeg, zlib, mesa, libGLU, xorg, ffmpeg }:
+
+
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/geometry/netgen-mesher/default.nix:4]:
```suggestion
{ stdenv, opencascade, fetchFromGitHub, cmake, python3, tcl, tk, libjpeg, zlib, mesa, libGLU, xorg, ffmpeg }:

stdenv.mkDerivation rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**